ops/ops/interface_tls_certificates/requires.py

```diff
@@ -92,15 +101,29 @@ def client_certs(self) -> List[Certificate]:
         certs_json = getattr(self._data, field, "{}")
         certs_data = json.loads(certs_json)
         return [
-            Certificate(cert_type="client", common_name=common_name, **cert)
-            for common_name, cert in certs_data.items()
+            Certificate(
+                cert_type="client",
+                common_name=common_name,
+                cert=self.build_chain(cert_data.get("cert")),
+                key=cert_data.get("key"),
+            )
+            for common_name, cert_data in certs_data.items()
         ]
 
     @property
     def client_certs_map(self) -> Mapping[str, Certificate]:
         """Certificate instances by their `common_name`."""
         return {cert.common_name: cert for cert in self.client_certs}
 
+    def build_chain(self, cert) -> str:
+        """Build a certificate chain. This will include the provided
+        client/server cert, along with additional intermediate certificates
+        that are needed to connect the client/server cert back to the root CA.
+        """
+        if self.chain:
+            cert += "\n" + self.chain
+        return cert
```

Comment:
I'm okay with this... but i wonder if it would have been any easier to bake this into the `Certificate` class as some kind of copy constructor

the following suggestion may not be right.  

```python
class Certificate(BaseModel):
    """Represent a Certificate."""

    cert_type: StrictStr
    common_name: StrictStr
    cert: StrictStr
    key: StrictStr

    def add_chain(self, chain: Optional[str]) -> 'Certificate':
        if not chain:
            return self
        return Certificate(
            cert_type=self.cert_type, 
            common_name=self.cert_type, 
            cert=self.cert+"\n"+chain, 
            key=self.key,
        )
```

Reply:
I moved the cert chain building to a new `Certificate.__init__` method. Hopefully that captures your intent here, what do you think?